File: sdk/agid-go/agid.go
package agid

const Base32Alphabet = "0123456789ABCDEFGHJKMNPQRSTVWXYZ"
const PrefixLength = 2
const HashLength = 10
const TotalLength = 12

type Result struct {
	ID   string
	Lat  float64
	Lon  float64
	Face int
}

type Bounds struct {
	MinLat float64
	MaxLat float64
	MinLon float64
	MaxLon float64
}

func Encode(lat float64, lon float64) (Result, error) {
	return Result{}, ErrNotImplemented
}

func Decode(id string) (Result, error) {
	return Result{}, ErrNotImplemented
}

func CellBounds(id string) (Bounds, error) {
	return Bounds{}, ErrNotImplemented
}

File: sdk/agid-zig/src/agid.zig
pub const base32_alphabet = "0123456789ABCDEFGHJKMNPQRSTVWXYZ";
pub const prefix_length = 2;
pub const hash_length = 10;
pub const total_length = 12;

pub const Result = struct {
    id: [total_length]u8,
    lat: f64,
    lon: f64,
    face: ?u8,
};

pub const Bounds = struct {
    minLat: f64,
    maxLat: f64,
    minLon: f64,
    maxLon: f64,
};

pub fn encode(lat: f64, lon: f64) !Result {
    _ = lat;
    _ = lon;
    return error.NotImplemented;
}

pub fn decode(id: []const u8) ?Result {
    _ = id;
    return null;
}

pub fn cellBounds(id: []const u8) !Bounds {
    _ = id;
    return error.NotImplemented;
}
